transforms/universal/pdf2md/python/src/pdf2md_transform.py

```diff
@@ -0,0 +1,346 @@
+            return [], {}
+
+# class PdfToMdRuntime(DefaultRayTransformRuntime):
+#     """
+#     Ingest to Parquet runtime support
+#     """
+
+#     def __init__(self, params: dict[str, Any]):
+#         """
+#         Create filter runtime
+#         :param params: parameters, that should include
+#             ingest_supported_langs_file_key: supported languages file
+#             ingest_detect_programming_lang_key: whether to detect programming language
+#             ingest_domain_key: domain
+#             ingest_snapshot_key: snapshot
+#         """
+#         super().__init__(params)
+
+#     def get_transform_config(
+#         self,
+#         data_access_factory: DataAccessFactoryBase,
+#         statistics: ActorHandle,
+#         files: list[str],
+#     ) -> dict[str, Any]:
+#         """
+#         Set environment for filter execution
+#         :param data_access_factory - data access factory
+#         :param statistics - reference to the statistics object
+#         :param files - list of files to process
+#         :return: dictionary of filter init params
+#         """
+
+#         # This cannot yet be done, because the converter is not easily serializable
+
+#         # self.models_dir = Path(self.params.get(ingest_modelsdir_key, "scratch"))
+#         # print("Downloading models")
+#         # logger.info("Downloading models")
+#         # DocumentConverter.download_models(
+#         #     artifacts_path=self.models_dir,
+#         #     artifacts_url=DocumentConverter.artifacts_urls["standard"],
+#         # )
+#         # print("Initializing converter")
+#         # logger.info("Initializing converter")
+#         # converter = DocumentConverter(artifacts_path=self.models_dir)
+#         # converter_refs = ray.put(converter)
+
+#         # logger.debug(f"Placed converter into Ray object storage under reference {converter_refs=}")
+#         # return {ingest_converter_refname: converter_refs} | self.params
+
+#         return self.params
+
+
+class PdfToMdTransformConfiguration(TransformConfiguration):
+    """
+    Provides support for configuring and using the associated Transform class include
+    configuration with CLI args and combining of metadata.
+    """
+
+    def __init__(self):
+        super().__init__(
+            name=shortname,
+            transform_class=PdfToMdTransform,
+            # remove_from_metadata=[ingest_data_factory_key],
+        )
+        self.daf = None
+
+    def add_input_params(self, parser: ArgumentParser) -> None:
+        """
+        Add Transform-specific arguments to the given parser.
+        By convention a common prefix should be used for all mutator-specific CLI args
+        (e.g, noop_, pii_, etc.)
+        """
+        parser.add_argument(
+            f"--{pdf2md_modelsdir_key}",
+            type=str,
+            help="Path where to download models",
+            default="scratch",
```

Comment:
All of these defaults need to also be implemented in the transform initializer.  Probably should define constants that hold the defaults and then use those instead of the literals in both places.

Reply:
The transform initializer has the same values. indeed, we can improve by using some literals

Reply:
literals are now in place and also used in the kfp defaults